File: src/main/java/com/thegame/business/service/TimerService.java
package com.thegame.business.service;

import com.thegame.business.dto.TimerUpdateRequestDTO;
import com.thegame.business.repository.TimerRepository;
import org.springframework.stereotype.Service;

import java.time.LocalDateTime;
import java.util.List;

@Service
public class TimerService {

    private final TimerRepository timerRepository;

    public TimerService(TimerRepository timerRepository) {
        this.timerRepository = timerRepository;
    }

    public List<TimerRepository.BuildingTimerByVillageResponse> getBuildingTimerByVillageId(Long villageId) {
        return timerRepository.getBuildingTimerByVillageId(villageId);
    }

    public void updateBuildingTimerByVillageId(TimerUpdateRequestDTO request) {
        LocalDateTime updateTime = LocalDateTime.now().plusSeconds(request.getDuration());
        timerRepository.updateBuildingTimerByVillageId(
                request.getVillageId(),
                request.getTimerTypeId(),
                request.getObjectTypeId(),
                updateTime
        );
    }
}

File: src/main/java/com/thegame/business/service/TimerService.kt
package com.thegame.business.service

import com.thegame.business.model.Resource
import com.thegame.business.repository.ResourceRepository
import com.thegame.business.repository.TimerRepository
import org.springframework.stereotype.Service

@Service
class TimerService(private val timerRepository: TimerRepository) {

    fun getBuildingTimerByVillageId(villageId: Long): List<TimerRepository.BuildingTimerByVillageResponse> {
        return timerRepository.getBuildingTimerByVillageId(villageId)
    }

}